main/src/ca/uwaterloo/flix/api/lsp/provider/GotoProvider.scala

```diff
@@ -27,74 +29,129 @@ object GotoProvider {
+    case _ => None
+  }
+
+  private def isReal(x: AnyRef): Boolean = x match {
+    case TypedAst.Trait(_, _, _, _, _, _, _, _, _, loc) =>  loc.isReal
+    case TypedAst.Instance(_, _, _, _, _, _, _, _, _, loc) => loc.isReal
+    case TypedAst.Sig(_, _, _, loc) => loc.isReal
+    case TypedAst.Def(_, _, _, loc) => loc.isReal
+    case TypedAst.Enum(_, _, _, _, _, _, _, loc) => loc.isReal
+    case TypedAst.Struct(_, _, _, _, _, _, _, loc) => loc.isReal
+    case TypedAst.RestrictableEnum(_, _, _, _, _, _, _, _, loc) => loc.isReal
+    case TypedAst.TypeAlias(_, _, _, _, _, _, loc) => loc.isReal
+    case TypedAst.AssocTypeSig(_, _, _, _, _, _, loc) => loc.isReal
+    case TypedAst.AssocTypeDef(_, _, _, _, _, loc) => loc.isReal
+    case TypedAst.Effect(_, _, _, _, _, loc) => loc.isReal
+    case TypedAst.Op(_, _, loc) => loc.isReal
+    case exp: TypedAst.Expr => exp.loc.isReal
+    case pat: TypedAst.Pattern => pat.loc.isReal
+    case TypedAst.RestrictableChoosePattern.Wild(_, loc) => loc.isReal
+    case TypedAst.RestrictableChoosePattern.Var(_, _, loc) => loc.isReal
+    case TypedAst.RestrictableChoosePattern.Tag(_, _, _, loc) => loc.isReal
+    case TypedAst.RestrictableChoosePattern.Error(_, loc) => loc.isReal
+    case p: TypedAst.Predicate => p.loc.isReal
+    case TypedAst.Binder(sym, _) => sym.loc.isReal
+    case TypedAst.Case(_, _, _, loc) => loc.isReal
+    case TypedAst.StructField(_, _, loc) => loc.isReal
+    case TypedAst.RestrictableCase(_, _, _, loc) => loc.isReal
+    case TypedAst.Constraint(_, _, _, loc) => loc.isReal
+    case TypedAst.ConstraintParam(_, _, loc) => loc.isReal
+    case TypedAst.FormalParam(_, _, _, _, loc) => loc.isReal
+    case TypedAst.PredicateParam(_, _, loc) => loc.isReal
+    case TypedAst.JvmMethod(_, _, _, _, _, loc) => loc.isReal
+    case TypedAst.CatchRule(_, _, _) => true
+    case TypedAst.HandlerRule(_, _, _) => true
+    case TypedAst.TypeMatchRule(_, _, _) => true
+    case TypedAst.SelectChannelRule(_, _, _) => true
+    case TypedAst.TypeParam(_, _, loc) => loc.isReal
+    case TypedAst.ParYieldFragment(_, _, loc) => loc.isReal
+
+    case SymUse.AssocTypeSymUse(_, loc) => loc.isReal
+    case SymUse.CaseSymUse(_, loc) => loc.isReal
+    case SymUse.DefSymUse(_, loc) => loc.isReal
+    case SymUse.EffectSymUse(_, loc) => loc.isReal
+    case SymUse.LocalDefSymUse(_, loc) => loc.isReal
+    case SymUse.OpSymUse(_, loc) => loc.isReal
+    case SymUse.RestrictableCaseSymUse(_, loc) => loc.isReal
+    case SymUse.RestrictableEnumSymUse(_, loc) => loc.isReal
+    case SymUse.SigSymUse(_, loc) => loc.isReal
+    case SymUse.StructFieldSymUse(_, loc) => loc.isReal
+    case SymUse.TraitSymUse(_, loc) => loc.isReal
+
+    case TraitConstraint(_, _, loc) => loc.isReal
+    case TraitConstraint.Head(_, loc) => loc.isReal
+
+    case AssocTypeConstructor(_, loc) => loc.isReal
+    case EqualityConstraint(_, _, _, loc) => loc.isReal
+
+    case _: Symbol => true
```

Comment:
What symbol is not covered by the above?

Reply:
Now that I think about it again, none. `x` should never be a Symbol at all, really, since the we're filtering things on the stack of the StackConsumer and the visitor never consumes a symbol (only ever a symbol occurrrence). So it should be safe to remove that line, which I'll do